system/Database/SQLSRV/PreparedQuery.php

```diff
@@ -80,19 +73,17 @@ public function _execute(array $data): bool
             $this->parameters[$key] = $value;
         }
 
-        $this->result = sqlsrv_execute($this->statement);
-
-        return (bool) $this->result;
+        return sqlsrv_execute($this->statement);
     }
 
     /**
-     * Returns the result object for the prepared query.
+     * Returns the statement resource for the prepared query.
```

Comment:
```suggestion
     * Returns the statement resource for the prepared query or false on failure.
```

Reply:
That would not be correct. Resource stays a resource even if the query execution fails.

Reply:
>Returns a statement resource on success and false if an error occurred.

sqlsrv_prepare https://www.php.net/manual/en/function.sqlsrv-prepare.php

Reply:
Something like this?
> Returns the statement resource for the prepared query or false when preparing failed.

Reply:
@sclubricants Okay, you're right. I forgot that in theory we can call `_getResult()` before we call `_execute()`. Thanks.

@kenjis That is even better. Thanks.